crates/core/src/sql/execute.rs

```diff
@@ -227,12 +225,11 @@ pub(crate) mod tests {
     #[test]
     fn test_select_catalog() -> ResultTest<()> {
         let (db, _, _tmp_dir) = create_data(1)?;
-        let mut tx = db.begin_tx();
+        let tx = db.begin_tx();
         let schema = db.schema_for_table(&tx, ST_TABLES_ID).unwrap().into_owned();
-
+        db.rollback_read_tx(&ExecutionContext::internal(db.address()), tx);
```

Comment:
If feels weird why this context is needed and the chance the `db.address()` is different.

Reply:
Sorry, I don't think I understood this comment well, can you please eloborate more?
 
though ctx is there only for the sake that `rollback_read_tx` expects it? Can it be done in better way?

Reply:
I mean that the context of the rollback looks like it can be different from the `db.begin_tx()`. Maybe I'm misreading the API but why is it not just `rollback_read_tx()`?

Reply:
Agree, It feels weird and may be 'ctx' should be binded to `tx` lifecycle but right now, code is following https://github.com/clockworklabs/SpacetimeDB/blob/master/crates/core/src/db/datastore/traits.rs#L69-#L71 current implmented trait. I feel usage of ctx needs to be fixed for datastore in general as there are incosistencies between methods expecting ctx.
I think most of it is out of scope of this PR and should be done separately.